pkg/cache/clusterqueue.go

```diff
@@ -341,18 +347,42 @@ func (c *ClusterQueue) deleteWorkload(w *kueue.Workload) {
 	// we only increase the AllocatableResourceGeneration cause the add of workload won't make more
 	// workloads fit in ClusterQueue.
 	c.AllocatableResourceGeneration++
+
 	delete(c.Workloads, k)
-	reportAdmittedActiveWorkloads(wi.ClusterQueue, len(c.Workloads))
+	c.reportActiveWorkloads()
+}
+
+func (c *ClusterQueue) reportActiveWorkloads() {
+	metrics.AdmittedActiveWorkloads.WithLabelValues(c.Name).Set(float64(c.admittedWorkloadsCount()))
+	metrics.ReservingActiveWorkloads.WithLabelValues(c.Name).Set(float64(len(c.Workloads)))
+}
+
+func (c *ClusterQueue) admittedWorkloadsCount() int {
+	count := 0
+	for _, wi := range c.Workloads {
```

Comment:
keep a counter instead
Otherwise, for each add/remove Workload, we are iterating over all workloads.

Reply:
done